tests/Console/ConfigurationResolverTest.php

```diff
@@ -944,6 +944,61 @@ public function testResolveCommandLineInputOverridesDefault()
         $this->assertSame('json', $resolver->getReporter()->getFormat());
     }
 
+    /**
+     * @param string $expected
+     * @param mixed  $differConfig
+     *
+     * @dataProvider provideDifferCases
+     */
+    public function testResolveDiffer($expected, $differConfig)
+    {
+        $resolver = new ConfigurationResolver(
+            $this->config,
+            array('diff' => $differConfig),
+            ''
+        );
+
+        $this->assertInstanceOf($expected, $resolver->getDiffer());
```

Comment:
This test does not guarantee that the differ instance uses the correct mode. Not sure how we could do that though.